library/src/commonMain/kotlin/com/badoo/reaktive/observable/CombineLatest.kt

```diff
@@ -0,0 +1,231 @@
+    source6: Observable<T6>,
+    mapper: (T1, T2, T3, T4, T5, T6) -> R
+): Observable<R> =
+    listOf(source1, source2, source3, source4, source5, source6)
+        .combineLatest { values ->
+            @Suppress("UNCHECKED_CAST")
+            mapper(values[0] as T1, values[1] as T2, values[2] as T3, values[3] as T4, values[4] as T5, values[5] as T6)
+        }
+
+fun <T1, T2, T3, T4, T5, T6, T7, R> combineLatest(
+    source1: Observable<T1>,
+    source2: Observable<T2>,
+    source3: Observable<T3>,
+    source4: Observable<T4>,
+    source5: Observable<T5>,
+    source6: Observable<T6>,
+    source7: Observable<T7>,
+    mapper: (T1, T2, T3, T4, T5, T6, T7) -> R
+): Observable<R> =
+    listOf(source1, source2, source3, source4, source5, source6, source7)
+        .combineLatest { values ->
+            @Suppress("UNCHECKED_CAST")
+            mapper(
+                values[0] as T1,
+                values[1] as T2,
+                values[2] as T3,
+                values[3] as T4,
+                values[4] as T5,
+                values[5] as T6,
+                values[6] as T7
+            )
+        }
+
+fun <T1, T2, T3, T4, T5, T6, T7, T8, R> combineLatest(
+    source1: Observable<T1>,
+    source2: Observable<T2>,
+    source3: Observable<T3>,
+    source4: Observable<T4>,
+    source5: Observable<T5>,
+    source6: Observable<T6>,
+    source7: Observable<T7>,
+    source8: Observable<T8>,
+    mapper: (T1, T2, T3, T4, T5, T6, T7, T8) -> R
+): Observable<R> =
+    listOf(source1, source2, source3, source4, source5, source6, source7, source8)
+        .combineLatest { values ->
+            @Suppress("UNCHECKED_CAST")
+            mapper(
+                values[0] as T1,
+                values[1] as T2,
+                values[2] as T3,
+                values[3] as T4,
+                values[4] as T5,
+                values[5] as T6,
+                values[6] as T7,
+                values[7] as T8
+            )
+        }
+
+fun <T1, T2, T3, T4, T5, T6, T7, T8, T9, R> combineLatest(
+    source1: Observable<T1>,
+    source2: Observable<T2>,
+    source3: Observable<T3>,
+    source4: Observable<T4>,
+    source5: Observable<T5>,
+    source6: Observable<T6>,
+    source7: Observable<T7>,
+    source8: Observable<T8>,
+    source9: Observable<T9>,
+    mapper: (T1, T2, T3, T4, T5, T6, T7, T8, T9) -> R
+): Observable<R> =
+    listOf(source1, source2, source3, source4, source5, source6, source7, source8, source9)
+        .combineLatest { values ->
+            @Suppress("UNCHECKED_CAST")
+            mapper(
+                values[0] as T1,
+                values[1] as T2,
+                values[2] as T3,
+                values[3] as T4,
+                values[4] as T5,
+                values[5] as T6,
+                values[6] as T7,
+                values[7] as T8,
+                values[8] as T9
+            )
+        }
+
+fun <T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, R> combineLatest(
```

Comment:
It will require making function inline which will produce incorrect stack traces if mapper throws an exception. Casting is actually safe in these cases.

Reply:
Also there is a limitation, the following code won't compile:
```
fun <T> foo(v: T) {
    bar(v)
}

inline fun <reified T> bar(v: T) {
}
```